diff --git a/buffer.js b/buffer.js
new file mode 100644
index 0000000000000000000000000000000000000000..abd9fc4b68c9e64fa7d6330ece69990fd2a7c833
--- /dev/null
+++ b/buffer.js
@@ -0,0 +1 @@
+module.exports.Buffer = typeof window === 'undefined' ? globalThis.Buffer : window.Buffer;
\ No newline at end of file
diff --git a/hash.js b/hash.js
index 013537a9747eecd0eeb16f8c95f4695617cdd5cf..a9842f86fd161371acf6a464251c497c07269e47 100644
--- a/hash.js
+++ b/hash.js
@@ -1,4 +1,4 @@
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 // prototype class for hash functions
 function Hash (blockSize, finalSize) {
diff --git a/sha.js b/sha.js
index 50c4fa801bad8fd31ddd816da4dc85e17bfc8e13..eff8e1510cb83d81a66e04e30c45ea4b4ad11211 100644
--- a/sha.js
+++ b/sha.js
@@ -8,7 +8,7 @@
 
 var inherits = require('inherits')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var K = [
   0x5a827999, 0x6ed9eba1, 0x8f1bbcdc | 0, 0xca62c1d6 | 0
diff --git a/sha1.js b/sha1.js
index cabd747ce8ec9f94985a0cfea7157174b8de9a15..5fab609397a80a61bc43fbe63af8cfe4dd9cbfe1 100644
--- a/sha1.js
+++ b/sha1.js
@@ -9,7 +9,7 @@
 
 var inherits = require('inherits')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var K = [
   0x5a827999, 0x6ed9eba1, 0x8f1bbcdc | 0, 0xca62c1d6 | 0
diff --git a/sha224.js b/sha224.js
index 35541e57584dcda89d731e125768b1c266f200c7..ae2a0ed142066cb7dec7a6d9e0b367e8a1d3af7b 100644
--- a/sha224.js
+++ b/sha224.js
@@ -9,7 +9,7 @@
 var inherits = require('inherits')
 var Sha256 = require('./sha256')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var W = new Array(64)
 
diff --git a/sha256.js b/sha256.js
index 342e48aeeef9462e72542e00062ed202b623a98b..2be7413f08997d63bd9a6793b40d4416657ba1c2 100644
--- a/sha256.js
+++ b/sha256.js
@@ -8,7 +8,7 @@
 
 var inherits = require('inherits')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var K = [
   0x428A2F98, 0x71374491, 0xB5C0FBCF, 0xE9B5DBA5,
diff --git a/sha384.js b/sha384.js
index afc85e51f6294890e77e62af925e73085a550d23..f33a671d8049cb937055362fda28c41d770e5c20 100644
--- a/sha384.js
+++ b/sha384.js
@@ -1,7 +1,7 @@
 var inherits = require('inherits')
 var SHA512 = require('./sha512')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var W = new Array(160)
 
diff --git a/sha512.js b/sha512.js
index fb28f2f6be8c45c9ac01acecc3fdb2a4bf069626..58393d5b6d98e3d9703ec28e11674d21102c524a 100644
--- a/sha512.js
+++ b/sha512.js
@@ -1,6 +1,6 @@
 var inherits = require('inherits')
 var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var K = [
   0x428a2f98, 0xd728ae22, 0x71374491, 0x23ef65cd,
diff --git a/test/vectors.js b/test/vectors.js
index 48a646ef691a2f3df0ecb4049cd716b5b4809132..250f2603a8ad11c17ffb131aec70c78f56cbb811 100644
--- a/test/vectors.js
+++ b/test/vectors.js
@@ -1,7 +1,7 @@
 var tape = require('tape')
 var vectors = require('hash-test-vectors')
 // var from = require('bops/typedarray/from')
-var Buffer = require('safe-buffer').Buffer
+var { Buffer } = require('./buffer')
 
 var createHash = require('../')
 